### ISSUE TITLE
Severe Lag - Fixed by ditching Twitch app? (release version)

### ISSUE THREAD

**Issue by SithVixenX** [scalda/FoolCraft#558]
Hello,

I've been watching Guude stream this modpack and love it, so decided to try it. When I installed an instance via Twitch I was getting insane amounts of lag. My CPU was pushed, ram pushed, every step caused a lag spike. I allocated 6G of ram. Minimal fixing. I set my graphics card to force use my CUDA processors, it fixed it more, but I still got spikes and eventually after a half hour it would just freeze. During this time I couldn't have anything else open, Chrome in particular would lag like a dying dragon.

So I saw someone mention how to move a different modpack into just plain old Minecraft, no Twitch launcher. I moved the mods folder, downloaded the appropriate version of Forge and it runs like a dream. However, several things are broken. I notice I now get a globe when loading in, achievements are active now, there are many more mobs, can't craft wooden tools, veinminer is not working (at least not for hand.)

I'm assuming I should move the configuration files from Twitch's folder as well? Would this fix the issues? Would this suddenly cause a renewal of the lag?

I'm also posting here as a food for thought, maybe this weird occurrence might inspire previously unthought of solutions to lag issues? I don't know.

I run GeForce GTX 970, allocated 6 of 8 gigs of ram, i5 processor (I know that sucks), 64bit Win7.

Any thoughts? Thanks for compiling this modpack, it looks like tons of fun. :)

**Comment by scalda** [comment]
Hi what version of the pack are you running?

**Comment by SitVixenX** [comment]
1.33 in both cases. I attempted 1.4.5 using Twitch app earlier tonight, but it worsened the lag so I went back.

**Comment by ajdjyt** [comment]
Can we get a log from each way you tried to play the modpack.
and also how big is your page file is it on on an SSD or an HDD and what is your CPUs clock speed?          and while playing what is your cached RAM at? and what is your RAM usage when your system is idling? 
this might seem like im asking for a lot but im asking for it for a good reason

**Comment by scalda** [comment]
also try turning your mipmap settings off in the video settings tab and make sure vsync is off aswell

**Comment by SitVixenX** [comment]
CPU clock: i5 2300 at 2.80GHz, don't think it's ever been overlocked tho my friend might have screwed with it once upon a time, I simply don't recall. Gonna say no to be safe.

Page File: Total for all drives - 8173. MC is on C, no SSDs (I wish).

Cached Ram: In Task Manager cashed says 1101. Avail 6131, free 5059. Total 8173.

Ram when Idling: According to Task Manager, physical memory is sitting at 24-25%, CPU 0%. 

* 

I'll turn off mipmap and vsync as well. :)  Logs coming. Thank you!

**Comment by SitVixenX** [comment]
Log files...hope this is what you meant. :) The one labelled latestF.zip is the one from the Twitch app.

[latestF.zip](https://github.com/scalda/FoolCraft/files/1088434/latestF.zip)
[latest.zip](https://github.com/scalda/FoolCraft/files/1088436/latest.zip)

**Comment by ajdjyt** [comment]
Uhh... Does it say 6131 avail and free 5059 while playing ? that means mc or java is low on priority or something is wrong cause the pack is set to use 5.4 GB of ram and what you told me says that it is using none  And the 2 GB RAM is used by windows and the pack for me on 8 gigs and an i3 @2.00Ghz uses 4.7 to 5.4 Gigs of RAM ,your cached RAM seems fine though and your page file is big enough

**Comment by ajdjyt** [comment]
Yay A broken recipe
[10:07:15] [Client thread/ERROR]: Found a broken crafting recipe.
mezz.jei.util.BrokenCraftingRecipeException: net.minecraft.item.crafting.ShapedRecipes@160b5f76
Outputs:
net.minecraft.item.ItemStack: [[1x (null)]]
Inputs:
net.minecraft.item.ItemStack: [[1xitem.mw_SteelPlate@0 mw:steelplate], [1xitem.mw_SteelPlate@0 mw:steelplate], [1xitem.mw_SteelPlate@0 mw:steelplate], [1xitem.mw_SteelIngot@0 mw:steelingot], [1xtile.button@32767 minecraft:stone_button], [1xitem.mw_SteelIngot@0 mw:steelingot], [], [], [1xitem.mw_SteelPlate@0 mw:steelplate]]
Caused by: java.lang.NullPointerException: Null item in itemStack
	at mezz.jei.util.StackHelper.getSubtypes(StackHelper.java:219) ~[StackHelper.class:?]
	at mezz.jei.util.StackHelper.getAllSubtypes(StackHelper.java:292) ~[StackHelper.class:?]
	at mezz.jei.util.StackHelper.getAllSubtypes(StackHelper.java:270) ~[StackHelper.class:?]
	at mezz.jei.plugins.vanilla.ingredients.ItemStackHelper.expandSubtypes(ItemStackHelper.java:24) ~[ItemStackHelper.class:?]
	at mezz.jei.util.Ingredients.setOutput(Ingredients.java:62) ~[Ingredients.class:?]
	at mezz.jei.plugins.vanilla.crafting.ShapedRecipesWrapper.getIngredients(ShapedRecipesWrapper.java:35) ~[ShapedRecipesWrapper.class:?]
	at mezz.jei.RecipeRegistry.getIngredients(RecipeRegistry.java:288) ~[RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.addRecipeUnchecked(RecipeRegistry.java:272) ~[RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.addRecipe(RecipeRegistry.java:242) [RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.addRecipe(RecipeRegistry.java:201) [RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.addRecipes(RecipeRegistry.java:184) [RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.<init>(RecipeRegistry.java:102) [RecipeRegistry.class:?]
	at mezz.jei.util.ModRegistry.createRecipeRegistry(ModRegistry.java:209) [ModRegistry.class:?]
	at mezz.jei.JeiStarter.start(JeiStarter.java:52) [JeiStarter.class:?]
	at mezz.jei.ProxyCommonClient.restartJEI(ProxyCommonClient.java:136) [ProxyCommonClient.class:?]
	at mezz.jei.ProxyCommonClient.access$000(ProxyCommonClient.java:37) [ProxyCommonClient.class:?]
	at mezz.jei.ProxyCommonClient$1.func_110549_a(ProxyCommonClient.java:105) [ProxyCommonClient$1.class:?]
	at net.minecraft.client.resources.SimpleReloadableResourceManager.func_110544_b(SimpleReloadableResourceManager.java:132) [bxi.class:?]
	at net.minecraft.client.resources.SimpleReloadableResourceManager.func_110541_a(SimpleReloadableResourceManager.java:113) [bxi.class:?]
	at net.minecraft.client.Minecraft.func_110436_a(Minecraft.java:755) [bcx.class:?]
	at net.minecraft.client.Minecraft$16.run(Minecraft.java:2677) [bcx$8.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_25]
	at net.minecraft.client.Minecraft.func_152343_a(Minecraft.java:3018) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_152344_a(Minecraft.java:3040) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_175603_A(Minecraft.java:2673) [bcx.class:?]
	at net.minecraft.client.settings.GameSettings.onGuiClosed(GameSettings.java:1288) [bcz.class:?]
	at net.minecraft.client.gui.GuiVideoSettings.func_146281_b(GuiVideoSettings.java:116) [bgb.class:?]
	at net.minecraft.client.Minecraft.func_147108_a(Minecraft.java:936) [bcx.class:?]
	at net.minecraft.client.gui.GuiVideoSettings.func_146284_a(GuiVideoSettings.java:68) [bgb.class:?]
	at net.minecraft.client.gui.GuiScreen.func_73864_a(GuiScreen.java:460) [bft.class:?]
	at net.minecraft.client.gui.GuiVideoSettings.func_73864_a(GuiVideoSettings.java:76) [bgb.class:?]
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:548) [bft.class:?]
	at net.minecraft.client.gui.GuiVideoSettings.func_146274_d(GuiVideoSettings.java:57) [bgb.class:?]
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:517) [bft.class:?]
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1700) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371) [bcx.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:124) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_25]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_25]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_25]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_25]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
[10:07:15] [Client thread/ERROR]: Found a broken crafting recipe.
mezz.jei.util.BrokenCraftingRecipeException: net.minecraft.item.crafting.ShapedRecipes@2e13cf6d
Outputs:
net.minecraft.item.ItemStack: [[32x (null)]]
Inputs:
net.minecraft.item.ItemStack: [[], [1xitem.mw_LeadIngot@0 mw:leadingot], [], [1xitem.mw_CopperIngot@0 mw:copperingot], [1xitem.sulphur@0 minecraft:gunpowder], [1xitem.mw_CopperIngot@0 mw:copperingot], [1xitem.mw_CopperIngot@0 mw:copperingot], [1xitem.sulphur@0 minecraft:gunpowder], [1xitem.mw_CopperIngot@0 mw:copperingot]]
Caused by: java.lang.NullPointerException: Null item in itemStack
	at mezz.jei.util.StackHelper.getSubtypes(StackHelper.java:219) ~[StackHelper.class:?]
	at mezz.jei.util.StackHelper.getAllSubtypes(StackHelper.java:292) ~[StackHelper.class:?]
	at mezz.jei.util.StackHelper.getAllSubtypes(StackHelper.java:270) ~[StackHelper.class:?]
	at mezz.jei.plugins.vanilla.ingredients.ItemStackHelper.expandSubtypes(ItemStackHelper.java:24) ~[ItemStackHelper.class:?]
	at mezz.jei.util.Ingredients.setOutput(Ingredients.java:62) ~[Ingredients.class:?]
	at mezz.jei.plugins.vanilla.crafting.ShapedRecipesWrapper.getIngredients(ShapedRecipesWrapper.java:35) ~[ShapedRecipesWrapper.class:?]
	at mezz.jei.RecipeRegistry.getIngredients(RecipeRegistry.java:288) ~[RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.addRecipeUnchecked(RecipeRegistry.java:272) ~[RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.addRecipe(RecipeRegistry.java:242) [RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.addRecipe(RecipeRegistry.java:201) [RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.addRecipes(RecipeRegistry.java:184) [RecipeRegistry.class:?]
	at mezz.jei.RecipeRegistry.<init>(RecipeRegistry.java:102) [RecipeRegistry.class:?]
	at mezz.jei.util.ModRegistry.createRecipeRegistry(ModRegistry.java:209) [ModRegistry.class:?]
	at mezz.jei.JeiStarter.start(JeiStarter.java:52) [JeiStarter.class:?]
	at mezz.jei.ProxyCommonClient.restartJEI(ProxyCommonClient.java:136) [ProxyCommonClient.class:?]
	at mezz.jei.ProxyCommonClient.access$000(ProxyCommonClient.java:37) [ProxyCommonClient.class:?]
	at mezz.jei.ProxyCommonClient$1.func_110549_a(ProxyCommonClient.java:105) [ProxyCommonClient$1.class:?]
	at net.minecraft.client.resources.SimpleReloadableResourceManager.func_110544_b(SimpleReloadableResourceManager.java:132) [bxi.class:?]
	at net.minecraft.client.resources.SimpleReloadableResourceManager.func_110541_a(SimpleReloadableResourceManager.java:113) [bxi.class:?]
	at net.minecraft.client.Minecraft.func_110436_a(Minecraft.java:755) [bcx.class:?]
	at net.minecraft.client.Minecraft$16.run(Minecraft.java:2677) [bcx$8.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_25]
	at net.minecraft.client.Minecraft.func_152343_a(Minecraft.java:3018) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_152344_a(Minecraft.java:3040) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_175603_A(Minecraft.java:2673) [bcx.class:?]
	at net.minecraft.client.settings.GameSettings.onGuiClosed(GameSettings.java:1288) [bcz.class:?]
	at net.minecraft.client.gui.GuiVideoSettings.func_146281_b(GuiVideoSettings.java:116) [bgb.class:?]
	at net.minecraft.client.Minecraft.func_147108_a(Minecraft.java:936) [bcx.class:?]
	at net.minecraft.client.gui.GuiVideoSettings.func_146284_a(GuiVideoSettings.java:68) [bgb.class:?]
	at net.minecraft.client.gui.GuiScreen.func_73864_a(GuiScreen.java:460) [bft.class:?]
	at net.minecraft.client.gui.GuiVideoSettings.func_73864_a(GuiVideoSettings.java:76) [bgb.class:?]
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:548) [bft.class:?]
	at net.minecraft.client.gui.GuiVideoSettings.func_146274_d(GuiVideoSettings.java:57) [bgb.class:?]
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:517) [bft.class:?]
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1700) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055) [bcx.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371) [bcx.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:124) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_25]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_25]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_25]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_25]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
[10:07:16] [Client thread/INFO]: Built    recipe registry in 1305 ms
[10:07:16] [Client thread/INFO]: Building item filter...

**Comment by SitVixenX** [comment]
When playing on the Twitch launcher (which I close after MC launcher pops up), avail sits at around 1000 or so, free hangs out at about 0. ;)  Sometimes jumps up a little, but not much more than 50.

**Comment by ajdjyt** [comment]
and the twitch launcher is also the same deal

**Comment by SitVixenX** [comment]
I turned off foliage in BOP for giggles. It seemed to help for a while, but then I had to take care of some phone stuff so I hit ESC and left it on the menu. When I came back it was lagging. Time definitely seems to be an issue, but I still get stuttering in the beginning anyway. It just gets worse.

Dxdiag says my video card has 4gig memory, I only have the one card. I'll grant you I prolly haven't updated drivers in a while. I suppose I could give that a go.

**Comment by ajdjyt** [comment]
and the driver update would help probably and your graphic card is ok .do you have optifine in the pack or betterfps if so then remove both mods and since the problem is a broken recipe only the devs of the pack can help you after this ,i would recommend reinstalling foolcraft from twitch by making a new instance of it

**Comment by SitVixenX** [comment]
I did install a new instance for this very test. No Optifine or BetterFPS. A broken recipe could cause that much of an issue? I assume others aren't getting this much lag.

**Comment by Jo-Mamma75** [comment]
@SitVixenX 
Ok.  Let's do this.  Get version 1.4.4 (1.4.5 has a couple of big issues we need to fix so 1.4.6 will hopefully be coming out shortly) via Twitch.  Follow the instructions that Scalda put in the Wiki about setting up the pack to run in MultiMC.  Make sure you allocate the appropriate amount of RAM.
The biggest RAM killer in the pack prior to version 1.4.4 was the "More Player Models" mod which was the cause of the run-away RAM usage experienced by so many people.
It sounds like you have a pretty decent computer so I can't possibly see why you would have so many issues.  It does sound like to me that when you attempted to setup the pack to use the Vanilla launcher the first time you missed copying some folders or files and that was why you had the problems you did.  That or they didn't download correctly via Curse/Twitch initially (which has been known to happen and is beyond our control).  If I can run this pack on a little crappy small form factor Dell OptiPlex with a Core i5, 8GB of RAM and a Radeon HD7570 then surely you can run it on your computer.

**Comment by SitVixenX** [comment]
Okay. What I've done since: Updated drivers. Got rid of all instances of Foolcraft, downloaded 1.4.5 via Twitch, allocated 6G and ran it. Turned off Vsync and Mipmap. Foolcraft ran pretty well with all other settings as I normally have them - fancy, 16 chunks, yadda. There was the occasional hesitation that I attribute to chunk loading, thing you get with any open world game. One medium sized lag spike without fries. Shut down was slow, but not horrible.

Did the MultiMC stuff as instructed, including memory settings. Was a bit laggy when I loaded in, dunno if that was due to it being a more treeish biome or what. I went in to turn vsync and mipmap off and had a catastrophic lag that led to the game being unresponsive. After the system calmed down I reran it, the settings had held so I loaded the world back. Was a bit slow loading in, thought it was gonna go unresponsive, but it evened out so I started running around. Seemed pretty decent, normal chunk hesitations, a few lag spikes. Shutting down was a little painful in terms of lag (like systemwide lag, Chrome was crying).

If I had to say, I think doing it the multiMC way is worse, which sounds counter intuitive, but in either event both were more playable than I was previously having (every 5 seconds a lag spike). In both I ran around one MC day and half the night in terms of how much time spent. RAM usage in F3 seemed to sit around 75%.

I'm noticing, however, even though I allocate 6G of ram, F3 will display the max allocated somewhere in the 5's. o_O At any rate, now it seems decently playable and for that I thank you guys all. :-D Now I can fart on my crops with the best of them.
